### PR TITLE
Adding support for --runtime-version flag for dapr init k8s

### LINKED ISSUE #307: Support --runtime-version flag for dapr init k8s
## In what area(s)?

> /area runtime

## Describe the feature

As of today we don't have support for `--runtime-version` flag when we are installing Dapr on Kubernetes. If no --runtime-version flag is provided it should download the latest release without filtering rc releases.

### PR DESCRIPTION
**Pull request by shalabhms**
# Description

To support honoring the --runtime-version flag for kubernetes init

## Issue reference

Closes #307

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: pkg/kubernetes/kubernetes.go [hunk 1]
```diff
@@ -20,16 +20,25 @@ import (
...
 	if err != nil {
 		return fmt.Errorf("can't connect to a Kubernetes cluster: %v", err)
 	}
 
-	version, err := cli_ver.GetLatestRelease(cli_ver.DaprGitHubOrg, cli_ver.DaprGitHubRepo)
-	if err != nil {
-		return fmt.Errorf("cannot get the manifest file: %s", err)
+	if version == daprLatestVersion {
+		var v string
+		v, err = cli_ver.GetLatestRelease(cli_ver.DaprGitHubOrg, cli_ver.DaprGitHubRepo)
```
Comment (amanbha) [inline pkg/kubernetes/kubernetes.go:36]: can it be refactored, this call should happen only when version is not latest, in all other cases it should just use the user provided version

Comment (shalabhms) [reply pkg/kubernetes/kubernetes.go:36]: > can it be refactored, this call should happen only when version is not latest, in all other cases it should just use the user provided version

The default version is always "latest" if not provided by user, so this block will get the latest release version number and use it to fetch respective latest versioned dapr-operator.yaml..

Comment (amanbha) [reply pkg/kubernetes/kubernetes.go:36]: in this case, will this call to "cli_ver.GetLatestRelease" be made when user provides a version?

File: pkg/kubernetes/kubernetes.go [hunk 2]
```diff
@@ -20,16 +20,25 @@ import (
 	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// Init deploys the Dapr operator
-func Init() error {
+const (
+	daprLatestVersion = "latest"
```
Comment (yaron2) [inline pkg/kubernetes/kubernetes.go:24]: Can this be exported? be made into `DaprLatestVersion` and then used from `init.go`:

`InitCmd.Flags().StringVarP(&runtimeVersion, "runtime-version", "", kubernetes.DaprLatestVersion, "The version of the Dapr runtime to install. for example: v0.1.0")`

File: pkg/kubernetes/kubernetes.go (hunk 1 above)
Comment (shalabhms) [reply pkg/kubernetes/kubernetes.go:36]: > in this case, will this call to "cli_ver.GetLatestRelease" be made when user provides a version?

no. user provided version will directly be used in fetching dapr_operator.yaml at Line 44.

File: pkg/kubernetes/kubernetes.go (hunk 2 above)
Comment (shalabhms) [reply pkg/kubernetes/kubernetes.go:24]: @yaron2 , if we export, then standalone package need to be changed as well. Also I prefer that it is better to keep init variable (runtimeVersion) separately and get the variable value ( default ) and then pass it over in packages and define "latest" as const in kubernetes.go which is as per current PR.

Comment (yaron2) [reply pkg/kubernetes/kubernetes.go:24]: ok

**Pull request merged**